src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -858,6 +867,88 @@ rebalance_table_shards(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * citus_rebalance_start rebalances the shards across the workers.
+ *
+ * SQL signature:
+ *
+ * citus_rebalance_start(
+ *     rebalance_strategy name DEFAULT NULL,
+ *     drain_only boolean DEFAULT false,
+ *     shard_transfer_mode citus.shard_transfer_mode default 'auto'
+ * ) RETURNS VOID
+ */
+Datum
+citus_rebalance_start(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+	List *relationIdList = NonColocatedDistRelationIdList();
+	Form_pg_dist_rebalance_strategy strategy =
+		GetRebalanceStrategy(PG_GETARG_NAME_OR_NULL(0));
+
+	PG_ENSURE_ARGNOTNULL(1, "drain_only");
+	bool drainOnly = PG_GETARG_BOOL(1);
+
+	PG_ENSURE_ARGNOTNULL(2, "shard_transfer_mode");
+	Oid shardTransferModeOid = PG_GETARG_OID(2);
+
+	RebalanceOptions options = {
+		.relationIdList = relationIdList,
+		.threshold = strategy->defaultThreshold,
+		.maxShardMoves = 10000000,
+		.excludedShardArray = construct_empty_array(INT4OID),
+		.drainOnly = drainOnly,
+		.rebalanceStrategy = strategy,
+		.improvementThreshold = strategy->improvementThreshold,
+	};
+	RebalanceTableShardsBackground(&options, shardTransferModeOid);
+	PG_RETURN_VOID();
```

Comment:
could we return the job ID? seems useful for automation

Reply:
definitely, was thinking about this already.

Reply:
Was trying to find how to return a named column, given that it might not be apparent what this function returns, but can't quickly find an example of that.

Simply returned the jobid as the result of this function.

Reply:
I think you'd need to return a record or SETOF record, so this is probably a lot simpler. In an interactive terminal, it should be somewhat obvious that it's the job ID from the fact that it matches the notices.